pulp_container/app/viewsets.py

```diff
@@ -857,64 +878,32 @@ class ContainerPushRepositoryViewSet(
+                "condition_expression": [
+                    "has_namespace_obj_perms:container.view_containerpushrepository or "
+                    "has_distribution_perms:container.view_containerpushrepository",
+                ],
             },
             {
                 "action": ["tag", "untag", "remove_image"],
                 "principal": "authenticated",
                 "effect": "allow",
-                "condition": [
-                    "has_namespace_or_obj_perms:container.modify_content_containerpushrepository",
-                    "has_namespace_or_obj_perms:container.view_containerpushrepository",
+                "condition_expression": [
+                    "has_namespace_obj_perms:container.modify_content_containerpushrepository or "
+                    "has_distribution_perms:container.modify_content_containerpushrepository",
                 ],
             },
             {
                 "action": ["update", "partial_update"],
                 "principal": "authenticated",
                 "effect": "allow",
                 "condition": [
-                    "has_namespace_or_obj_perms:container.change_containerpushrepository",
-                    "has_namespace_or_obj_perms:container.view_containerpushrepository",
-                ],
-            },
-        ],
-        "permissions_assignment": [
-            {
-                "function": "add_perms_to_distribution_group",
-                "parameters": {
-                    "group_type": "owners",
-                    "add_user_to_group": True,
-                },
-                "permissions": [
-                    "container.view_containerpushrepository",
-                    "container.modify_content_containerpushrepository",
-                    "container.change_containerpushrepository",
-                ],
-            },
-            {
-                "function": "add_perms_to_distribution_group",
-                "parameters": {
-                    "group_type": "collaborators",
-                    "add_user_to_group": False,
-                },
-                "permissions": [
-                    "container.view_containerpushrepository",
-                    "container.modify_content_containerpushrepository",
-                    "container.change_containerpushrepository",
-                ],
-            },
-            {
-                "function": "add_perms_to_distribution_group",
-                "parameters": {
-                    "group_type": "consumers",
-                    "add_user_to_group": False,
-                },
-                "permissions": [
-                    "container.view_containerpushrepository",
+                    "has_namespace_obj_perms:container.change_containerpushrepository or "
+                    "has_distribution_perms:container.change_containerpushrepository",
                 ],
             },
         ],
     }
+    LOCKED_ROLES = {}
```

Comment:
this is still TBD right?

Reply:
Is this the push-repo? Then no. I want to check all the permissions by the roles on the distribution. Because you cannot assign a role to the repo and the distribution tied together. (as in "will be considered a single role assignment")